python/ray/tune/analysis/experiment_analysis.py

```diff
@@ -875,3 +863,7 @@ def __init__(self, *args, **kwargs):
             "The `Analysis` class is being "
             "deprecated. Please use `ExperimentAnalysis` instead."
         )
+
+
+def _decode_checkpoint_from_experiment_state(cp: Union[str, dict]) -> dict:
+    return json.loads(cp, cls=TuneFunctionDecoder) if isinstance(cp, str) else cp
```

Comment:
When are the checkpoints strings?

Reply:
They can be either JSON strings or already parsed dicts. I believe the former is the default state when loading from disk

Reply:
Are they strings as long as `trials` are not set?